diagnostics-extension/src/dpsl.types.ts

```diff
@@ -0,0 +1,241 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview dpsl.* types definitions.
+ */
+
+// dpsl.telemetry.* type definitions
+
+/**
+ * Response message containing Backlight Info
+ */
+export interface BacklightInfoObject {
+  path: string;
+  maxBrightness: number;
+  brightness: number;
+}
+
+/**
+ * Response message containing Backlight Info
+ */
+export type BacklightInfo = BacklightInfoObject[];
+
+/**
+ * Response message containing Battery Info
+ */
+export interface BatteryInfo {
+  cycleCount: bigint;
+  voltageNow: number;
+  vendor: string;
+  serialNumber: string;
+  chargeFullDesign: number;
+  chargeFull: number;
+  voltageMinDesign: number;
+  modelName: string;
+  chargeNow: number;
+  currentNow: number;
+  technology: string;
+  status: string;
+  manufactureDate: string;
+  temperature: bigint;
+}
+
+/**
+ * Response message containing Bluetooth Info
+ */
+export interface BluetoothInfoObject {
+  name: string;
+  address: string;
+  powered: boolean;
+  numConnectedDevices: number;
+}
+
+/**
+ * Response message containing Bluetooth Info
+ */
+export type BluetoothInfo = BluetoothInfoObject[];
+
+/**
+ * Response message containing VPD Info
+ */
+export interface VpdInfo {
+  skuNumber: string;
+  serialNumber: string;
+  modelName: string;
+}
+
+/**
+ * Response message containing CPU Info
+ */
+export interface CpuInfo {
+  numTotalThreads: number;
+  architecture: string;
+  physicalCpus: Array<unknown>; // Replace unknown
+}
+
+/**
+ * Response message containing Fan Info
+ */
+export interface FanInfo {
+  speedRpm: number;
+}
```

Comment:
FanInfo should be an array.

https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/resources/dpsl/dpsl_types.js;drc=043c35d129e66b4ebe45509c4070a39a10fbd06f;l=80